src/Components/ServiceScene/Breakdowns/Patterns/PatternNameLabel.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useRef, useState } from 'react';
+import { AdHocFiltersVariable, sceneGraph } from '@grafana/scenes';
+import { Spinner, Toggletip } from '@grafana/ui';
+import { getLokiDatasource } from 'services/scenes';
+import { IndexScene } from 'Components/IndexScene/IndexScene';
+import { VAR_FILTERS } from 'services/variables';
+import { buildLokiQuery } from 'services/query';
+import { PatternFieldLabelStats } from './PatternFieldLabelStats';
+import { LogLabelStatsModel, TimeRange } from '@grafana/data';
+import { reportAppInteraction, USER_EVENTS_ACTIONS, USER_EVENTS_PAGES } from 'services/analytics';
+
+interface PatternNameLabelProps {
+  exploration: IndexScene;
+  pattern: string;
+}
+
+const LINE_LIMIT = 2000;
+
+export const PatternNameLabel = ({ exploration, pattern }: PatternNameLabelProps) => {
+  const patternIndices = extractPatternIndices(pattern);
+  const [stats, setStats] = useState<LogLabelStatsModel[][] | undefined>(undefined);
```

Comment:
```suggestion
  const [stats, setStats] = useState<LogLabelStatsModel[][]>([]);
```

This would simplify the code a bit, as you don't need to check if `stats` is undefined.